ZenPacks/zenoss/OpenStackInfrastructure/modeler/plugins/zenoss/OpenStackInfrastructure.py

```diff
@@ -111,46 +103,47 @@ def collect(self, device, unused):
 
         results['nova_url'] = yield client.nova_url()
 
-        result = yield client.flavors(detailed=True, is_public=None)
+        result = yield client.nova_flavors(is_public=True)
         results['flavors'] = result['flavors']
+        result = yield client.nova_flavors(is_public=False)
+        results['flavors'].extend(result['flavors'])
         log.debug('flavors: %s\n' % str(results['flavors']))
 
-        result = yield client.images(detailed=True)
+        result = yield client.nova_images()
         results['images'] = result['images']
         log.debug('images: %s\n' % str(results['images']))
 
-        result = yield client.hypervisors(detailed=False,
-                                          hypervisor_match='%',
-                                          servers=True)
+        result = yield client.nova_hypervisors(hypervisor_match='%',
+                                               servers=True)
         results['hypervisors'] = result['hypervisors']
         log.debug('hypervisors: %s\n' % str(results['hypervisors']))
 
-        result = yield client.hypervisors(detailed=True)
+        result = yield client.nova_hypervisorsdetailed()
         results['hypervisors_detailed'] = result['hypervisors']
         log.debug('hypervisors_detailed: %s\n' % str(results['hypervisors']))
 
-        result = yield client.servers(detailed=True, search_opts={'all_tenants': 1})
+        # get hypervisor details for each individual hypervisor
+        results['hypervisor_details'] = {}
+        for hypervisor in results['hypervisors']:
+            result = yield client.nova_hypervisors(
+                hypervisor_id=hypervisor['id'])
+            hypervisor_id = prepId("hypervisor-{0}".format(hypervisor['id']))
+            results['hypervisor_details'][hypervisor_id] = result['hypervisor']
+
```

Comment:
What is going on here?    I see no explanation for this change.  Why are you making all these extra API calls?

I see the comment below:    "if results['hypervisors_detailed'] did not give us hypervisor type, hypervisor version, try results['hypervisors_details']"

Need some more explanation of why you're incurring extra API calls here- If it is necessary on some versions, then I would suggest making it conditional, so that this extra stuff isn't done all the time.

Changes like this really should be tied to a specific jira with more explanation, since this changes existing behavior.